src/core_init_atmosphere/mpas_init_atm_thompson_aerosols.F

```diff
@@ -0,0 +1,753 @@
+! Copyright (c) 2024 The University Corporation for Atmospheric Research (UCAR).
+!
+! Unless noted otherwise source code is licensed under the BSD license.
+! Additional copyright and license information can be found in the LICENSE file
+! distributed with this code, or at http://mpas-dev.github.com/license.html
+!
+!=================================================================================================================
+ module mpas_init_atm_thompson_aerosols
+ use mpas_derived_types
+ use mpas_kind_types
+ use mpas_log
+ use mpas_dmpar
+ use mpas_pool_routines
+
+ use init_atm_read_met
+ use init_atm_hinterp
+ use init_atm_llxy
+ use init_atm_vinterp
+ use mpas_atmphys_date_time
+ use mpas_atmphys_utilities
+
+ implicit none
+ private
+ public:: init_atm_thompson_aerosols
+
+
+!mpas_init_atm_thompson_aerosols contains the subroutines needed for the interpolation of climatological
+!monthly-averaged hygroscopic ("water friendly") and nonhygroscopic ("ice friendly") aerosols used in the
+!Thompson parameterization of cloud microphysics with Gocart CCN and IN nucleation.
+!Laura D. Fowler (laura@ucar.edu) / 2024-04-10.
+
+
+ contains
+
+
+!=================================================================================================================
+ subroutine init_atm_thompson_aerosols(block,mesh,configs,diag,state)
+!=================================================================================================================
+
+!input arguments:
+ type (mpas_pool_type),intent(in):: configs
+ type(mpas_pool_type),intent(in):: diag
+
+!inout arguments:
+ type(block_type),intent(inout),target:: block 
+ type(mpas_pool_type),intent(inout)   :: mesh
+ type(mpas_pool_type),intent(inout)   :: state
+!local variables and pointers:
+ character (len=StrKIND),pointer:: config_start_time
+ character(len=StrKIND):: filename_gocart
+ character(len=StrKIND):: initial_date,mess
+
+ logical:: lexist
+
+!-----------------------------------------------------------------------------------------------------------------
+ call mpas_log_write('--- enter subroutine init_atm_gocart:')
```

Comment:
I'd propose removing this print statement as well as the `--- end subroutine init_atm_gocart.` print statement at the end of this subroutine.

Reply:
As suggested, commented out extra print statements.

Reply:
As suggested, moved the declaration of dimensions above that of variables.